### PR TITLE
Add missing UnitSystem to variant and EOSPAC

### PR DESCRIPTION
**Pull request by rbberger**
## PR Summary

- fixes formatting issue
- avoids C++17 usage warning in clang
- rewrites how variant type lists for Relativistic and UnitSystem are created, adding missing combinations
- also allows EOSPAC to make use of it
- add unit test for `UnitSystem<EOSPAC>`

@Yurlungur please take a closer look at what I had to do in `SpecificHeatFromDensityTemperature` to keep the scalar and vector functions consistent.

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

File: singularity-eos/eos/eos_eospac.hpp [hunk 1]
```diff
@@ -659,20 +659,17 @@ class EOSPAC : public EosBase<EOSPAC> {
...
 
+    const Real y = transform.y.is_set() ? (1.0 / transform.y.get()) : 1.0;
+    const Real f = transform.f.is_set() ? transform.f.get() : 1.0;
+
     portableFor(
         cname, 0, num, PORTABLE_LAMBDA(const int i) {
-          cvs[i] = std::max(cvs[i], 0.0); // Here we do something to the data!
+          cvs[i] =
+              f * y *
+              cvFromSesame(std::max(DEDT[i], 0.0)); // Here we do something to the data!
```
[the remark this answers is not in the document]
Comment (Yurlungur) [reply singularity-eos/eos/eos_eospac.hpp:672]: Either way I think we don't put the onus on this PR to change things if we decide to do that. We leave it for a later MR.

**Review by Yurlungur — CHANGES_REQUESTED** [review]
Overall looks good. Looks like the python bindings test is failing, though. Also you're missing a changelog. ;)

File: singularity-eos/eos/eos.hpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ namespace singularity {
 template <typename... Ts>
 using tl = singularity::detail::type_list<Ts...>;
 
-template <template <typename> typename... Ts>
+template <template <typename> class... Ts>
```
[the remark this answers is not in the document]
Comment (rbberger) [reply singularity-eos/eos/eos.hpp:58]: No, up to C++14 this is what was supported. Starting with C++17 you can write `typename...` here. Clang gives you a C++17 usage warning here.

[the remark this answers is not in the document]
Comment (rbberger) [reply singularity-eos/eos/eos.hpp:58]: See https://en.cppreference.com/w/cpp/language/parameter_pack

[the remark this answers is not in the document]
Comment (Yurlungur) [reply singularity-eos/eos/eos.hpp:58]: Huh. Good to know. I thought they were equivalent.

[the remark this answers is not in the document]
Comment (dholladay00) [reply singularity-eos/eos/eos.hpp:58]: I think xl also gave a warning there, in my mr #245 it has this change as well.

**Comment by rbberger** [comment]
@Yurlungur Added the CHANGELOG.md and fixed the Python bindings.

**Review by Yurlungur — APPROVED** [review]

File: singularity-eos/eos/eos_eospac.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mauneyc-LANL) [reply singularity-eos/eos/eos_eospac.hpp:672]: @Yurlungur see this from meeting

**Comment by Yurlungur** [comment]
Submitted tests to re-git. When they pass let's get this in.

**Pull request merged**